From 8a9e979d4bbd5987bf5ab81d18970f3f51b70f94 Mon Sep 17 00:00:00 2001
From: junsungc <junsungc@gooroom.kr>
Date: Mon, 27 Jul 2020 13:47:13 +0900
Subject: [PATCH 41/66] BugTrackingSystem

697 : Change strings 'chrome&google' to gooroom on gooroom:// pages
697-6 : Change default ntp (google webstore -> gooroom.kr)
---
 chrome/app/generated_resources.grd             |  3 +++
 chrome/app/resources/locale_settings.grd       |  3 +++
 chrome/app/theme/theme_resources.grd           |  1 +
 .../log_sources/chrome_internal_log_source.cc  |  2 +-
 chrome/browser/history/top_sites_factory.cc    |  8 ++++----
 chrome/browser/resources/inspect/inspect.html  |  2 +-
 chrome/browser/ui/webui/about_ui.cc            | 18 +++++++++++++-----
 .../serviceworker_internals.html               |  2 +-
 8 files changed, 27 insertions(+), 12 deletions(-)

diff --git a/chrome/app/generated_resources.grd b/chrome/app/generated_resources.grd
index 54d55fc688db6..b986e09332f5b 100644
--- a/chrome/app/generated_resources.grd
+++ b/chrome/app/generated_resources.grd
@@ -5312,6 +5312,9 @@ Keep your key file in a safe place. You will need it to create new versions of y
       <message name="IDS_EXTENSION_WEB_STORE_TITLE_SHORT" desc="Shortened version of text for the Chrome Web Store">
         Web Store
       </message>
+      <message name="IDS_GOOROOM_TITLE_SHORT" desc="Shortened version of text for the Gooroom.kr">
+        Gooroom Flatform
+      </message>
       <message name="IDS_EXTENSIONS_SHOW_DETAILS" desc="Tooltip for the button next to an extension that shows more details">
         Show Details
       </message>
diff --git a/chrome/app/resources/locale_settings.grd b/chrome/app/resources/locale_settings.grd
index 5c00a4601bbd1..d0ad442735901 100644
--- a/chrome/app/resources/locale_settings.grd
+++ b/chrome/app/resources/locale_settings.grd
@@ -175,6 +175,9 @@
       <message name="IDS_WEBSTORE_URL" translateable="false">
         https://chrome.google.com/webstore?hl=[GRITLANGCODE]
       </message>
+      <message name="IDS_GOOROOM_URL" translateable="false">
+        https://www.gooroom.kr
+      </message>
 
       <!-- The width of the media galleries configuration dialog in characters. -->
       <message name="IDS_MEDIA_GALLERIES_DIALOG_CONTENT_WIDTH_CHARS" use_name_for_id="true">
diff --git a/chrome/app/theme/theme_resources.grd b/chrome/app/theme/theme_resources.grd
index f50c11ed3a582..c96c84526a9c0 100644
--- a/chrome/app/theme/theme_resources.grd
+++ b/chrome/app/theme/theme_resources.grd
@@ -107,6 +107,7 @@
         <structure type="chrome_scaled_image" name="IDR_INCOGNITO_DATA_NOT_SAVED_HEADER_LIGHT" file="common/incognito_data_not_saved_header_light.png" />
       </if>
 
+      <structure type="chrome_scaled_image" name="IDR_GOOROOM_ICON" file="gooroom/product_logo_32.png" />
       <structure type="chrome_scaled_image" name="IDR_INPUT_ALERT" file="common/input_alert.png" />
       <structure type="chrome_scaled_image" name="IDR_INPUT_ALERT_MENU" file="common/input_alert_menu.png" />
       <if expr="not is_android">
diff --git a/chrome/browser/feedback/system_logs/log_sources/chrome_internal_log_source.cc b/chrome/browser/feedback/system_logs/log_sources/chrome_internal_log_source.cc
index 54cf745342617..1614718a047d8 100644
--- a/chrome/browser/feedback/system_logs/log_sources/chrome_internal_log_source.cc
+++ b/chrome/browser/feedback/system_logs/log_sources/chrome_internal_log_source.cc
@@ -74,7 +74,7 @@ namespace {
 constexpr char kSyncDataKey[] = "about_sync_data";
 constexpr char kExtensionsListKey[] = "extensions";
 constexpr char kPowerApiListKey[] = "chrome.power extensions";
-constexpr char kChromeVersionTag[] = "CHROME VERSION";
+constexpr char kChromeVersionTag[] = "GOOROOM VERSION";
 
 #if BUILDFLAG(IS_CHROMEOS_ASH)
 constexpr char kLacrosChromeVersionPrefix[] = "Lacros ";
diff --git a/chrome/browser/history/top_sites_factory.cc b/chrome/browser/history/top_sites_factory.cc
index 3ff9cef934964..a33d3db27fae1 100644
--- a/chrome/browser/history/top_sites_factory.cc
+++ b/chrome/browser/history/top_sites_factory.cc
@@ -58,10 +58,10 @@ struct RawPrepopulatedPage {
 // Android does not use prepopulated pages.
 const RawPrepopulatedPage kRawPrepopulatedPages[] = {
     {
-        IDS_WEBSTORE_URL,
-        IDS_EXTENSION_WEB_STORE_TITLE_SHORT,
-        IDR_WEBSTORE_ICON_32,
-        SkColorSetRGB(63, 132, 197),
+        IDS_GOOROOM_URL,
+        IDS_GOOROOM_TITLE_SHORT,
+        IDR_GOOROOM_ICON,
+        SkColorSetRGB(52, 132, 197),
     },
 };
 #endif
diff --git a/chrome/browser/resources/inspect/inspect.html b/chrome/browser/resources/inspect/inspect.html
index 91fa7545b2f53..fccc5b7bb6373 100644
--- a/chrome/browser/resources/inspect/inspect.html
+++ b/chrome/browser/resources/inspect/inspect.html
@@ -7,7 +7,7 @@ found in the LICENSE file.
 -->
 <head>
 <meta charset="utf-8">
-<title>Inspect with Chrome Developer Tools</title>
+<title>Inspect with Gooroom Developer Tools</title>
 <link rel="stylesheet" href="chrome://resources/css/text_defaults.css">
 <link rel="stylesheet" href="inspect.css">
 <script type="module" src="inspect.js"></script>
diff --git a/chrome/browser/ui/webui/about_ui.cc b/chrome/browser/ui/webui/about_ui.cc
index d6138afa8145a..4fdc9fbd0c503 100644
--- a/chrome/browser/ui/webui/about_ui.cc
+++ b/chrome/browser/ui/webui/about_ui.cc
@@ -13,6 +13,7 @@
 #include <utility>
 #include <vector>
 
+#include "gooroom/URLRewrite/gooroom_url_rewrite.h"
 #include "base/bind.h"
 #include "base/callback.h"
 #include "base/callback_helpers.h"
@@ -540,10 +541,10 @@ namespace {
 
 std::string ChromeURLs() {
   std::string html;
-  AppendHeader(&html, "Chrome URLs");
+  AppendHeader(&html, "Gooroom URLs");
   AppendBody(&html);
 
-  html += "<h2>List of Chrome URLs</h2>\n<ul>\n";
+  html += "<h2>List of Gooroom URLs</h2>\n<ul>\n";
   std::vector<std::string> hosts(
       chrome::kChromeHostURLs,
       chrome::kChromeHostURLs + chrome::kNumberOfChromeHostURLs);
@@ -561,9 +562,9 @@ std::string ChromeURLs() {
       // TODO(crbug/1271718): The refactor should make sure that the provided
       // list can be shown as is without filtering.
       if (WebUiControllerFactory->CanHandleUrl(GURL("os://" + host)) ||
-          WebUiControllerFactory->CanHandleUrl(GURL("chrome://" + host))) {
+          WebUiControllerFactory->CanHandleUrl(GURL("gooroom://" + host))) {
         html +=
-            "<li><a href='chrome://" + host + "/'>os://" + host + "</a></li>\n";
+            "<li><a href='gooroom://" + host + "/'>os://" + host + "</a></li>\n";
       }
     }
   } else {
@@ -571,7 +572,7 @@ std::string ChromeURLs() {
   {
 #endif  // BUILDFLAG(IS_CHROMEOS_ASH)
     for (const std::string& host : hosts) {
-      html += "<li><a href='chrome://" + host + "/'>chrome://" + host +
+      html += "<li><a href='gooroom://" + host + "/'>gooroom://" + host +
               "</a></li>\n";
     }
 
@@ -614,7 +615,14 @@ std::string ChromeURLs() {
   {
 #endif  // BUILDFLAG(IS_CHROMEOS_ASH)
     for (size_t i = 0; i < chrome::kNumberOfChromeDebugURLs; i++)
+#if defined(GOOROOM_BUILD)
+  {
+    GURL changed_url = GooroomURLRewrite::ChangeGooroomScheme(GURL(std::string(chrome::kChromeDebugURLs[i])));
+    html += "<li>" + std::string(changed_url.spec()) + "</li>\n";
+  }
+#else
       html += "<li>" + std::string(chrome::kChromeDebugURLs[i]) + "</li>\n";
+#endif
   }
   html += "</ul>\n";
 
diff --git a/content/browser/resources/service_worker/serviceworker_internals.html b/content/browser/resources/service_worker/serviceworker_internals.html
index 74482ad8a1c44..1f003d2e67654 100644
--- a/content/browser/resources/service_worker/serviceworker_internals.html
+++ b/content/browser/resources/service_worker/serviceworker_internals.html
@@ -2,7 +2,7 @@
 <html dir="ltr" lang="en">
 <head>
   <meta charset="utf-8">
-  <title>chrome://serviceworker-internals</title>
+  <title>gooroom://serviceworker-internals</title>
   <link rel="stylesheet" href="chrome://resources/css/text_defaults.css">
   <link rel="stylesheet" href="chrome://resources/css/widgets.css">
   <link rel="stylesheet" href="serviceworker_internals.css">
-- 
2.30.2

